neo/Network/RPC/RpcClient.cs

```diff
@@ -257,9 +260,14 @@ public RpcInvokeResult InvokeFunction(string address, string function, RpcStack[
         /// Returns the result after passing a script through the VM.
         /// This RPC call does not affect the blockchain in any way.
         /// </summary>
-        public RpcInvokeResult InvokeScript(byte[] script)
+        public RpcInvokeResult InvokeScript(byte[] script, params JObject[] extraParams)
```

Comment:
Why do we need `extraParams`?

Reply:
When we construct a transaction we need to calculate the system fee by calling ```InvokeScript```. With the ```extraParams``` we can pass the ```CheckWitness``` function when the script needs to check witness.

Reply:
I think it can be a part of `script`.

Reply:
The case ```invokescript``` in ```RpcServer``` accepts extra parameters now:
```
case "invokescript":
                    {
                        byte[] script = _params[0].AsString().HexToBytes();
                        CheckWitnessHashes checkWitnessHashes = null;
                        if (_params.Count > 1)
                        {
                            UInt160[] scriptHashesForVerifying = _params.Skip(1).Select(u => UInt160.Parse(u.AsString())).ToArray();
                            checkWitnessHashes = new CheckWitnessHashes(scriptHashesForVerifying);
                        }
                        return GetInvokeResult(script, checkWitnessHashes);
                    }
```

Reply:
I did not understand it perfectly, @chenquanyu.

The `extraParams` is like a bypass for forcing the result?

Reply:
You may refer to this PR https://github.com/neo-project/neo/pull/335

Reply:
That is true, I had forgot, @chenquanyu.
Thanks!

Reply:
We should rename it.

Reply:
Renamed the parameter to `scriptHashesForVerifying` for better understanding.
And I also changed the type to `UInt160[]`.